addons/docs/src/blocks/mdx.tsx

```diff
@@ -49,8 +49,16 @@ export const CodeOrSourceMdx: FC<CodeOrSourceMdxProps> = ({ className, children,
   );
 };
 
+function generateHrefWithHash(hash: string): string {
+  // eslint-disable-next-line no-undef
+  const location = window.parent ? window.parent.location : window.location;
```

Comment:
This is an illegal operation cross origin!

Reply:
which can happen when a storybook is composed in:
https://storybook.js.org/docs/react/sharing/storybook-composition#gatsby-focus-wrapper

Reply:
@ndelangen  Got any suggestions to update the location?

Reply:
BTW, your link seems to point to a header called "gatsby-focus-wrapper" but i don't see it in the page

Reply:
I think that the issue is not the check but the fact that we're not passing the correct URL
We are passing only the hash and not appending it to the already existing URL or replacing it with existing hash in the URL.

Reply:
When we call `addons.getChannel().emit(NAVIGATE_URL, url)`, where is the actual code that navigating? Maybe we will have access to the full URL there and we can change the logic there.

There's a `getBase` function in `router.tsx`

```js
const getBase = () => `${document.location.pathname}?`;
```

What are the implications of `document.location` vs `window.location`?